File: src/haven/purus/pbot/api/PBotFlowerMenu.java
package haven.purus.pbot.api;

import haven.FlowerMenu;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class PBotFlowerMenu {

	private FlowerMenu menu;

	PBotFlowerMenu(FlowerMenu menu) {
		this.menu = menu;
	}

	/**
	 * Close this flowermenu
	 */
	public void closeMenu() {
		this.menu.choose(null);
		this.menu.destroy();
	}

	/**
	 * Choose option
	 * @param name Exact name of the option
	 * @return whether the option was chosen or not
	 */
	public boolean choosePetal(String name) {
		for(FlowerMenu.Petal p : this.menu.opts) {
			if(p.name.equals(name)) {
				try {
					menu.choose(p);
					menu.destroy();
					return true;
				} catch(IndexOutOfBoundsException iee) {}
			}
		}
		return false;
	}

	/**
	 * Choose option by number
	 * @param num Option number 0-indexed
	 */
	public void choosePetal(int num) {
		menu.choose(menu.opts[num]);
		menu.destroy();
	}

	/**
	 * Get options of flowermenu
	 * @return Names of the petal options
	 */
	public List<String> getPetalNames() {
		return Arrays.stream(this.menu.opts)
				.map(opt -> opt.name)
				.collect(Collectors.toList());
	}
}

File: scripts/__pbot/PBotFlowerMenu.py
from typing import List

class PBotFlowerMenu(object):
    def __init__(self, menu):
        self.__menu = menu

    ## Close this flowermenu
    def close(self):
        self.__menu.closeMenu()

    ## Choose option
    # @param name Exact name of the option
    # @return true if the petal was chosen, false if not found
    def choose_petal(self, name: str) -> bool:
        return self.__menu.choosePetal(name)

    ## Choose option by number
    # @param num option number 0-indexed
    def choose_petal_num(self, num: int):
        self.__menu.choosePetal(num)

    ## Get options of this flowermenu
    # @return list of options
    def get_petal_names(self) -> List[str]:
        return list(self.__menu.getPetalNames())